base/strings/search.jl

```diff
@@ -166,35 +188,40 @@ function _search_bloom_mask(c)
 end
 
 _nthbyte(s::String, i) = codeunit(s, i)
-_nthbyte(a::Union{AbstractVector{UInt8},AbstractVector{Int8}}, i) = a[i]
+_nthbyte(a::Union{Vector{UInt8},Vector{Int8}}, i) = a[i]
+_nthbyte(t::AbstractVector, index) = t[firstindex(t) + (index-1)]
```

Comment:
I would normally write this as
```jl
t[index + (firstindex(t)-1)]
```
Either way, if `firstindex(t)` is a compile-time constant (as is the case for typical Julia arrays), then LLVM will simplify the indexing calculations.    But if `firstindex(t)` is *not* a compile-time constant, then separating out the index-independent calculation `firstindex(t)-1` might make it easier for the compiler to perform common-subexpression elimination.